tests/test_plugin_kraken.py

```diff
@@ -39,6 +44,12 @@ def test_kraken(mocker: Any) -> None:
 
     client: Exchange = plugin._client
 
+    mocker.patch.object(client, "load_markets").return_value = None
+    client.markets_by_id = {  # type: ignore
```

Comment:
no it wasn't a type declaration problem as it is already part of the instance. When I did: 
`mocker.patch.object(client, "markets_by_id", {...})`
It did not set the attribute the way I expected. 
So I realized I could set it myself and that's why I did it. 

If you know how to do it through mocker or another suggestion, please provide code, otherwise I think it's fine.

Reply:
How was it not set the way you expected? It was in a different format or something?

Reply:
When I set it, for example, in the form `{"XLTCUSD": Dict[str, str]}` the code received it in some other form because it didn't process it correctly. I'm not sure what mocker is doing underneath.

Reply:
@ndopencode, can you address this comment by @macanudo527? After that, I think we're ready to merge!

Reply:
I am trying to update my local environment to the recent merges. I think we added backports as I get that as an error running pytest tests. When I run the update command in my (windows) terminal

`.venv/bin/pip3 install -e ".[dev]"`

I get 

`'.venv' is not recognized as an internal or external command,
operable program or batch file.`

When I look inside the .venv/ folder, I don't see a bin folder. Did I get the update command wrong?

Reply:
That's strange, could you try pulling a new tree from scratch in a new directory and applying your diffs:
```
git clone https://github.com/eprbell/dali-rp2
cd dali-rp2
git fetch origin pull/114/head:pr-114; git checkout pr-114
```

Then (assuming you're not on Windows):
```
virtualenv -p python3 .venv
. .venv/bin/activate
.venv/bin/pip3 install -e '.[dev]'
```

Then see if it works:
```
pytest --tb=native --verbose
```

Reply:
I'm on Windows. It looks like I had things written wrong and needed to use `python -m pip install -e ".[dev]"` instead. This was in README.md.dev. I tried your instructions and they work for PR-114. However, for main I get errors associated with backports. When updating my dev using 
`python -m pip install -e ".[dev]"` 
I get: 

`ERROR: Failed building wheel for backports-datetime-fromisoformat
error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/`
I get this even after downloading and installing the following from: https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170 
VC_redist.x64.exe
VC_redist.x86.exe 

This seems like a new dependency added in the last few days. Does dali instructions need to be updated?

Reply:
Yes, backports-datetime-fromisoformat was introduced yesterday. I don't think instructions need to be updated: it sounds like you need to `git pull` on your development branch. Let  me know if that works.

Reply:
Another thing you can try (make sure to backup your directory first):
```
rm -rf $(VENV) .mypy_cache/ .pytest_cache/ build dist/ src/*.egg-info/
```
Then repeat [installation instructions](https://github.com/eprbell/dali-rp2/blob/main/README.dev.md#setup-on-windows-10).

Reply:
Ok. I was able to get it to work with the following. I did it in a fresh virtualenv. I had to do the following to get things setup. 

1. Download "vs_BuildTools.exe" from https://visualstudio.microsoft.com/visual-cpp-build-tools/
2. Install the following Microsoft packages: 

C++ Build Tools core features
C++ 2022 Redistributable Update
C++ core desktop features
MSVC v143 - VS 2022 C++ x64/x86 build tools
Windows 10 SDK (10.0.20348.0)

The first three were required. The fourth one was enabled by default. The last one was required. This built and installed `backports-datetime-fromisoformat`.